### ISSUE TITLE
[Request] Setup torrent before download start

### ISSUE THREAD

**Issue by SkipperQ93** [XITRIX/iTorrent#106]
It would be really nice if we can select files before the download starts. Also if we can select whether to enable sequential downloading as well.
Right now as soon as torrent starts to download, I pause it, change files selection, enable sequential download and then resume downloading. Only downside is that even if I am quick, it still starts downloading the files that are not needed, hence creating unnecessary files. 
I am sure this would be the issue with others as well. That’s why normally desktop torrent clients give this option of files selection before torrent starts to download. 
Thank you.

**Comment by XITRIX** [comment]
Are you downloading from magnet links? It you'll add torrent using .torrent file, app will ask you what files do you want to download.

**Comment by SkipperQ93** [comment]
Yes I just noticed that it’s fine with torrent files. How about adding this option with magnet URLs as well. 
Another suggestion would be to add a generalised option in settings to automatically start download or show dialog, like in many apps. 
And it would be great to add an option to enable sequential downloading along with file selection.

**Comment by XITRIX** [comment]
1. I think it will be hard to do with magnet link, cause app don't know when download will begin cause it should to find metadata first.
2. Do you mean like "Download all files automatically or ask me every time"?
3. I'll think about it, but now I am not sure where I could place such button to make it looks nice.

**Comment by SkipperQ93** [comment]
1 and 2 are related to magnet links. Well, if there is a problem in detecting the download start, then surely it will be an issue to show dialog. 
3. Sure, that’s up to you. Thanks.

**Comment by aidan-gibson** [comment]
Unsure if I should open new issue or just piggyback this one, but I recently had to manually uncheck like 300 files when I just wanted one. There's gotta be a UI remedy to this

**Comment by XITRIX** [comment]
@aidan-gibson, nice point, need to add "select all" button at least
